barretenberg/cpp/src/barretenberg/protogalaxy/decider_prover.cpp

```diff
@@ -0,0 +1,120 @@
+#include "decider_prover.hpp"
+#include "barretenberg/sumcheck/sumcheck.hpp"
+
+namespace proof_system::honk {
+
+/**
+ * Create DeciderProver_ from an accumulator.
+ *
+ * @param accumulator Relaxed instance (ϕ, ω' \vec{β*}, e*) whose proof we want to generate, produced by Protogalaxy
+ * folding prover
+ *
+ * @tparam a type of UltraFlavor
+ * */
+template <UltraFlavor Flavor>
+DeciderProver_<Flavor>::DeciderProver_(const std::shared_ptr<Instance>& inst,
+                                       const std::shared_ptr<CommitmentKey>& commitment_key,
+                                       const std::shared_ptr<Transcript>& transcript)
+    : accumulator(std::move(inst))
+    , transcript(transcript)
+    , commitment_key(commitment_key)
+{}
+
+/**
+ * @brief Add  ϕ, \vec{β*}, e* to the transcript.
+ */
+template <UltraFlavor Flavor> void DeciderProver_<Flavor>::execute_preamble_round()
+{
+    const auto accumulator_size = static_cast<uint32_t>(accumulator->instance_size);
+    const auto num_public_inputs = static_cast<uint32_t>(accumulator->public_inputs.size());
+    transcript->send_to_verifier("accumulator_size", accumulator_size);
+    transcript->send_to_verifier("public_input_size", num_public_inputs);
+
+    for (size_t i = 0; i < accumulator->public_inputs.size(); ++i) {
+        auto public_input_i = accumulator->public_inputs[i];
+        transcript->send_to_verifier("public_input_" + std::to_string(i), public_input_i);
+    }
+
+    transcript->send_to_verifier("eta", accumulator->relation_parameters.eta);
+    transcript->send_to_verifier("beta", accumulator->relation_parameters.beta);
+    transcript->send_to_verifier("gamma", accumulator->relation_parameters.gamma);
+    transcript->send_to_verifier("public_input_delta", accumulator->relation_parameters.public_input_delta);
+    transcript->send_to_verifier("lookup_grand_product_delta",
```

Comment:
Where is this computed? We don't want to rely on comments but some would definitely be helpful here

Reply:
I'll make it explicit that these were computed in the last round of folding, thanks